CedarJavaFFI/src/interface_result.rs

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "success")]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
+#[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
+/// As of this writing, this is only used by `json_parse_entity_uid()`.
+/// We could remove this and just specialize the output of `json_parse_entity_uid()`.
```

Comment:
Do we think we'll add more operations with this return type? If not, I'd lean towards specializing the output of `json_parse_entity_uid()`

Reply:
I think we'll add more operations in general, in the future.  I am unclear whether they will/should use this return type or not

Reply:
A fair question. Let's just leave it in the Java code for now. But for consistency with our other data types, I'm thinking that we just want it to be named `Result` or `Answer` (no "interface"). I'd also prefer not to have the internal/non-internal failure distinction, but I don't feel strongly.

Reply:
Hmm my comment about only used by `json_parse_entity_uid()` appears to be wrong.  I suspect though that post-793 and friends, in some error cases our code is returning the wrong error structure.  Java code probably needs more tests with invalid call JSON.

Reply:
I think it makes the most sense to just rename `InterfaceResult` to `Answer`, and leave the rest for followups.  I've done that in the latest revision of this PR